### PR TITLE
Opt Into Perf Logging

### LINKED ISSUE #5900: Opt into and capture perf logging in second-stage bootstrap builds
Dogfood #5861 in our repo to catch regressions and also capture hopefully-useful telemetry from the build.

### PR DESCRIPTION
**Pull request by BenVillalobos**
Fixes #5900

### Context
It's time we opt into performance logging: https://github.com/dotnet/msbuild/pull/5861

### Changes Made
- PerformanceLogEventListener creates the `DOTNET_PERFLOG_DIR` directory if it doesn't exist.
- Opt into perf logging in `eng\cibuild_bootstrapped_msbuild.cmd` and the `.sh` equivalent.
- Added a powershell script that runs after windows/core and windows/full bootstrapped_msbuild scripts. This script runs `taskkill /f /im msbuild.exe` and `taskkill /f /im VBCSCompiler.exe` (these tasks were seen as holding a lock on some generated perf files). If we don't kill these tasks before trying to copy over the perf logs, it will fail the entire perf-log upload because the out of proc node (node mode 1) is still holding a lock on the file.

### Testing
We _should_ see perf logs under `artifacts/log/<configuration>/PerformanceLogs`

### Notes
Don't review commit by commit 😬 needless to say, this PR should be squashed.

### CODE REVIEW

**Review by brianrob — APPROVED** [review]
LGTM.  Thanks!

**Comment by BenVillalobos** [comment]
Note to self: Contact dnceng and ask the failures in copying perf logging artifacts. 

https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4565213&view=results shows a failure that it couldn't find the files to copy. Is BuildConfiguration not set at this point in `vsts-dotnet.yml`?
`env: { DOTNET_PERFLOG_DIR: '$(Build.SourcesDirectory)\artifacts\log\$(BuildConfiguration)\perf' }`

**Comment by BenVillalobos** [comment]
Perf logs can fail to publish if msbuild nodes hang onto them. 
```
Fail to upload 'D:\a\1\s\artifacts\log\Debug\PerformanceLogs\perf-7724-f1fd5808709141d399664b2c8ca3a4f7.log' due to 'The process cannot access the file 'D:\a\1\s\artifacts\log\Debug\PerformanceLogs\perf-7724-f1fd5808709141d399664b2c8ca3a4f7.log' because it is being used by another process.'.
System.IO.IOException: The process cannot access the file 'D:\a\1\s\artifacts\log\Debug\PerformanceLogs\perf-7724-f1fd5808709141d399664b2c8ca3a4f7.log' because it is being used by another process.
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.FileContainerServer.UploadAsync(IAsyncCommandContext context, Int32 uploaderId, Boolean uploadToBlob, CancellationToken token)
Async Command End: Upload Artifact
##[error]File upload failed even after retry.
```

Not the end of the world. Looks like CoreOnWindows published its perf logs. CoreOnMac and CoreOnLinux don't seem to have captured any perflogs. @brianrob is there any reason they shouldn't be?

**Comment by brianrob** [comment]
@BenVillalobos, I would expect that if the environment variable is set to a writeable location, then it should just work.  Are you able to reproduce this locally?  If not, it might be worth it to add some `printf` style debugging, and then you can look at the log to see what happened.

**Comment by BenVillalobos** [comment]
The environment variable is being set properly, logs are being created and populated. The publish step is running into issues because of a lock on a log file.

Uploading perf logs status
----
Windows full: consistently fails
Windows core: sometimes fails
macos core: works
linux core: works

There's typically a lock on a log file, which is strange because msbuild runs with `/nr` so the processes should be terminated when the build completes.
```
Fail to upload 'D:\a\1\s\artifacts\log\Debug\PerformanceLogs\perf-6700-556fe571db2441b38b38519d6a040b36.log' due to 'The process cannot access the file 'D:\a\1\s\artifacts\log\Debug\PerformanceLogs\perf-6700-556fe571db2441b38b38519d6a040b36.log' because it is being used by another process.'.
System.IO.IOException: The process cannot access the file 'D:\a\1\s\artifacts\log\Debug\PerformanceLogs\perf-6700-556fe571db2441b38b38519d6a040b36.log' because it is being used by another process.
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at Microsoft.VisualStudio.Services.Agent.Worker.Build.FileContainerServer.UploadAsync(IAsyncCommandContext context, Int32 uploaderId, Boolean uploadToBlob, CancellationToken token)
```

I tried to get the publish task to allow multiple retries, but it looks like that parameter just isn't being used: https://github.com/microsoft/azure-pipelines-tasks/issues/11451

Some ideas:
A. Set up the script such that a failed file upload doesn't prevent the rest from uploading
B. Ensure perflogeventlistener is disposed of earlier and therefore drops the lock on the file? /cc: @brianrob 


Total side note: I noticed our cibuilds still try to publish netcoreapp2.1 logs, when our logs are net5.0 now. I don't think those publish steps are necessary.

**Comment by brianrob** [comment]
Interesting.  Ideally we don't make upload of the log conditional, as this is a nice functional test of the whole system.

I think it could be worth scripting a check to see what process is locking the file.  It's going to be an MSBuild process, but perhaps the process is still alive even though we think it shouldn't be.  The file name includes the PID for the process that owns the log.  Could you fetch the command line for the PID and print it to the console?

Since this is limited to Windows, you could do something like:
```
Get-WmiObject -Query "SELECT CommandLine FROM Win32_Process WHERE ProcessID = <PID>"
```

File: .vsts-dotnet-ci.yml
Comment (Forgind) [inline .vsts-dotnet-ci.yml:8]: I'm confused by why all these other changes. Like why did you move them around? And did you just move them around? It's hard to tell what the other changes are.

File: src/MSBuild.UnitTests/PerfLog_Tests.cs
Comment (Forgind) [inline src/MSBuild.UnitTests/PerfLog_Tests.cs:77]: Why did this test used to exist? Seems a little odd to validate that we don't actually capture logs...

File: src/MSBuild/PerformanceLogEventListener.cs
Comment (Forgind) [inline src/MSBuild/PerformanceLogEventListener.cs:74]: Are Console.WriteLines the right way to log? If so, are they meant to be user-facing? If so, shouldn't we localize them? Either way, why so many exclamation points? Also, this should be much simpler if it's actually needed:
```suggestion
                Console.WriteLine(e.Message + "Inner Exception: " + e.InnerException);
```

File: .vsts-dotnet-ci.yml
Comment (BenVillalobos) [reply .vsts-dotnet-ci.yml:8]: The changes to this file should be viewed by commits. I have a commit where all I did was move the only non-bootstrap build to the top so that it runs first.

File: src/MSBuild.UnitTests/PerfLog_Tests.cs
Comment (BenVillalobos) [reply src/MSBuild.UnitTests/PerfLog_Tests.cs:77]: The test validated that if you set a path that didn't exist, nothing would happen. I changed the logic of the eventlistener to create the directory, so the test wasn't necessary anymore.

File: src/MSBuild/PerformanceLogEventListener.cs
Comment (BenVillalobos) [reply src/MSBuild/PerformanceLogEventListener.cs:74]: This was part of a debug commit that will be reverted when the PR is ready to go in.

**Comment by BenVillalobos** [comment]
Unfortunately it looks like passing parameters to this task (ie, retry 5 times) isn't supported, as its deprecated. And moving to a newer task would require updating the pool of machines we're running on. I'm not sure how involved that process would be, but I assume it's a yak to be shaved.

**Comment by BenVillalobos** [comment]
@brianrob 
Here's the script:

```powershell
      Get-ChildItem -Path "artifacts\log\Debug\PerformanceLogs\*" | Where-Object {$_.Extension -eq '.log'} | ForEach-Object {
        $s = $_.BaseName.Split('-')[1];
        Write-Host "Checking Process ID: $s"
        Get-WmiObject -Query "SELECT CommandLine FROM Win32_Process WHERE ProcessID = $s"
      }
```
Details from the process that's locking the file


### Windows Full
Looks legit.
```
Checking Process ID: 1096


__GENUS          : 2
__CLASS          : Win32_Process
__SUPERCLASS     : 
__DYNASTY        : 
__RELPATH        : 
__PROPERTY_COUNT : 1
__DERIVATION     : {}
__SERVER         : 
__NAMESPACE      : 
__PATH           : 
CommandLine      : D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\MSBuild.exe /nologo 
                   /nodemode:1 /nodeReuse:true /low:false
PSComputerName   : 

```

And 

This one may be a newer process that spawned and happened to take the same PID.
```
Checking Process ID: 3232
__GENUS          : 2
__CLASS          : Win32_Process
__SUPERCLASS     : 
__DYNASTY        : 
__RELPATH        : 
__PROPERTY_COUNT : 1
__DERIVATION     : {}
__SERVER         : 
__NAMESPACE      : 
__PATH           : 
CommandLine      : \??\C:\Windows\system32\conhost.exe 0x4
PSComputerName   : 

```
Looks legit.
```
Checking Process ID: 5300
__GENUS          : 2
__CLASS          : Win32_Process
__SUPERCLASS     : 
__DYNASTY        : 
__RELPATH        : 
__PROPERTY_COUNT : 1
__DERIVATION     : {}
__SERVER         : 
__NAMESPACE      : 
__PATH           : 
CommandLine      : "D:\a\1\s\artifacts\bin\Microsoft.Build.Tasks.UnitTests\Debug\net472\Roslyn\VBCSCompiler.exe" 
                   "-pipename:NuYm3XeVYobbHIPpGrn+XHTco0Zkr6qZXvQrQA55Cqw"
PSComputerName   : 

```


### Windows Core
Looks like it caught the previous powershell script running? Or it's catching itself? Likely catching itself.
```
Checking Process ID: 2668


__GENUS          : 2
__CLASS          : Win32_Process
__SUPERCLASS     : 
__DYNASTY        : 
__RELPATH        : 
__PROPERTY_COUNT : 1
__DERIVATION     : {}
__SERVER         : 
__NAMESPACE      : 
__PATH           : 
CommandLine      : "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe"  -NoLogo -NoProfile -NonInteractive 
                   -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\ce3ad429-b778-4ca5-a071-347039ee03c9.ps1'"
PSComputerName   : 

Checking Process ID: 2668
__GENUS          : 2
__CLASS          : Win32_Process
__SUPERCLASS     : 
__DYNASTY        : 
__RELPATH        : 
__PROPERTY_COUNT : 1
__DERIVATION     : {}
__SERVER         : 
__NAMESPACE      : 
__PATH           : 
CommandLine      : "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe"  -NoLogo -NoProfile -NonInteractive 
                   -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\ce3ad429-b778-4ca5-a071-347039ee03c9.ps1'"
PSComputerName   : 

```

This one looks legit.
```
Checking Process ID: 3252
__GENUS          : 2
__CLASS          : Win32_Process
__SUPERCLASS     : 
__DYNASTY        : 
__RELPATH        : 
__PROPERTY_COUNT : 1
__DERIVATION     : {}
__SERVER         : 
__NAMESPACE      : 
__PATH           : 
CommandLine      : "D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Roslyn\VBCSCompiler.exe" 
                   "-pipename:8Zt8SJ9zCljCMApPK4hTcAue0XXCpOUXsyHtUwkrKSQ"
PSComputerName   : 

```

```
Checking Process ID: 5184
__GENUS          : 2
__CLASS          : Win32_Process
__SUPERCLASS     : 
__DYNASTY        : 
__RELPATH        : 
__PROPERTY_COUNT : 1
__DERIVATION     : {}
__SERVER         : 
__NAMESPACE      : 
__PATH           : 
CommandLine      : D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\MSBuild.exe /nologo 
                   /nodemode:1 /nodeReuse:true /low:false
PSComputerName   : 

```

**Comment by BenVillalobos** [comment]
I have a `killnodereuse` script I often run that does this whenever processes are hanging or colliding with each other.

It's quick and dirty, but we could insert a script that does:
```
taskkill /f /im msbuild.exe
taskkill /f /im dotnet.exe
taskkill /f /im vbcscompiler.exe
taskkill /f /im omnisharp.exe
```

After the build script runs. Or change the timing where a `/nr:true` msbuild node will die off? Not sure exactly where this happens.

**Comment by brianrob** [comment]
Yeah, you could kill off the processes to make sure that you're able to copy the files.  What I find interesting is that some of these processes shouldn't be generating files.  VBCSCompiler.exe shouldn't be unless it is loading msbuild.  Same with PowerShell.  Basically, if a process doesn't load msbuild, then it shouldn't be creating a log file.

It would be interesting to kill these processes off, see which ones had to be killed, and what was in the log file.  Perhaps there are long-lived processes that are living beyond the CI run and we just didn't know that until now.

**Comment by BenVillalobos** [comment]
Ready for review. Recommend reviewing the diff rather than by commits. Lots of test commits used. The final diff looks a lot cleaner. I think it's good to go unless we have other ideas for getting around the failed perf log upload.

File: .vsts-dotnet-ci.yml
Comment (Forgind) [inline .vsts-dotnet-ci.yml:38]: This very much looks like a workaround. Why are there hanging processes in the first place? Can we fix that? Also, this makes it a little cleaner:
```suggestion
      taskkill /f /im msbuild.exe /im vbcscompiler.exe
```

**Review by Forgind — COMMENTED** [review]
I don't love the taskkill, but this LGTM other than that!

File: .vsts-dotnet-ci.yml
Comment (KirillOsenkov) [reply .vsts-dotnet-ci.yml:38]: I didn't know you can pass /im multiple times!

**Review by brianrob — APPROVED** [review]

File: .vsts-dotnet-ci.yml
Comment (brianrob) [reply .vsts-dotnet-ci.yml:38]: One thing to clarify here: My expectation is that there were always processes hanging around.  I don't believe that this change modifies process lifetime.  You could try running `dotnet build-server shutdown` to kill the compiler server.  One thing to confirm - are we getting full perf logs from these runs, or are they truncated because of the call to `taskkill`?

File: .vsts-dotnet-ci.yml
Comment (BenVillalobos) [reply .vsts-dotnet-ci.yml]: Noting that I reverted this change. My goal was to change the order these jobs run in, but they're listed in alphabetical order anyway 😕

**Comment by BenVillalobos** [comment]
Notes from PR review:

We want to get Rainer's take on it.
Some options:
1. Take the change as-is, killing the tasks like this is hacky but doesn't impact any user scenario.
2. Reduce the time it takes for OOP nodes to shut down after an `nr:false` build.
3. Update the task we're using to allow for multiple retries
	- The task we're using, PublishBuildArtifacts, [has a bug](https://github.com/microsoft/azure-pipelines-tasks/issues/11451) that doesn't allow arguments to be passed. The newer task, `PublishPipelineArtifacts` doesn't take any arguments and potentially doesn't use robocopy so I'm not sure if there are even arguments to pass.

File: .vsts-dotnet-ci.yml
Comment (BenVillalobos) [reply .vsts-dotnet-ci.yml:43]: If we kill the processes, the file that they generate isn't completed:

```
[2021-04-05T19:37:33.6451866Z] Event=Microsoft-Build/MSBuildExeStart ProcessID=7688 ThreadID=1	 commandLine="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\MSBuild.exe /nologo /nodemode:1 /nodeReuse:true /low:false" 
[2021-04-05T19:37:34.0831946Z] Event=Microsoft-Build/RequestThreadProcStart ProcessID=7688 ThreadID=11	 
[2021-04-05T19:37:34.0872085Z] Event=Microsoft-Build/BuildProjectStart ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.0962009Z] Event=Microsoft-Build/LoadDocumentStart ProcessID=7688 ThreadID=11	 fullPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.1042039Z] Event=Microsoft-Build/LoadDocumentStop ProcessID=7688 ThreadID=11	 fullPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.1072967Z] Event=Microsoft-Build/ParseStart ProcessID=7688 ThreadID=11	 projectFileName="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.1142095Z] Event=Microsoft-Build/ParseStop ProcessID=7688 ThreadID=11	 projectFileName="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.3042076Z] Event=Microsoft-Build/TargetStart ProcessID=7688 ThreadID=11	 targetName="msbuild" 
[2021-04-05T19:37:34.3582095Z] Event=Microsoft-Build/LoadDocumentStart ProcessID=7688 ThreadID=11	 fullPath="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Microsoft.Common.tasks" 
[2021-04-05T19:37:34.3602064Z] Event=Microsoft-Build/LoadDocumentStop ProcessID=7688 ThreadID=11	 fullPath="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Microsoft.Common.tasks" 
[2021-04-05T19:37:34.3602064Z] Event=Microsoft-Build/ParseStart ProcessID=7688 ThreadID=11	 projectFileName="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Microsoft.Common.tasks" 
[2021-04-05T19:37:34.3621959Z] Event=Microsoft-Build/ParseStop ProcessID=7688 ThreadID=11	 projectFileName="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Microsoft.Common.tasks" 
[2021-04-05T19:37:34.4402082Z] Event=Microsoft-Build/TargetStop ProcessID=7688 ThreadID=11	 targetName="msbuild" 
[2021-04-05T19:37:34.4442095Z] Event=Microsoft-Build/BuildProjectStop ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" targets="msbuild" 
[2021-04-05T19:37:34.4462038Z] Event=Microsoft-Build/RequestThreadProcStop ProcessID=7688 ThreadID=11	 
[2021-04-05T19:37:34.5532097Z] Event=Microsoft-Build/RequestThreadProcStart ProcessID=7688 ThreadID=11	 
[2021-04-05T19:37:34.5532097Z] Event=Microsoft-Build/BuildProjectStart ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.5571985Z] Event=Microsoft-Build/TargetStart ProcessID=7688 ThreadID=11	 targetName="msbuild" 
[2021-04-05T19:37:34.5612210Z] Event=Microsoft-Build/TargetStop ProcessID=7688 ThreadID=11	 targetName="msbuild" 
[2021-04-05T19:37:34.5612210Z] Event=Microsoft-Build/BuildProjectStop ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" targets="msbuild" 
[2021-04-05T19:37:34.5612210Z] Event=Microsoft-Build/RequestThreadProcStop ProcessID=7688 ThreadID=11	 
[2021-04-05T19:40:57.3420042Z] Event=Microsoft-Build/RequestThreadProcStart ProcessID=7688 ThreadID=11	 
[2021-04-05T19:40:57.3420042Z] Event=Microsoft-Build/BuildProjectStart ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmpe37eba57a280484fb0a6015308bebaff.tmp" 
[2021-04-05T19:40:57.3430119Z] Event=Microsoft-Build/LoadDocumentStart ProcessID=7688 ThreadID=11	 fullPath="C:\Users\VssAdministrator\App
```
Notice how it ends on `LoadDocumentStart` and has no `MSBuildExeStop`

I'll try `dotnet build-server` shutdown, which I didn't realize was an option!

Comment (Forgind) [reply .vsts-dotnet-ci.yml]: If certain events hadn't fired, wouldn't that still mean those events wouldn't fire as it tries to get out of the build as fast as possible?

Comment (BenVillalobos) [reply .vsts-dotnet-ci.yml]: Right. I'd like to get this in a state that is "good enough" if we want to see this perf data and act on it. Even if it's not the perfect solution. Part of me suspects digging into this node issue could take longer than expected. I've been digging into it without much result so far. The current idea is to set the MSBUILDDISABLENODEREUSE env var to 1 and see if the node will realize it needs to die.

**Comment by BenVillalobos** [comment]
Current status is setting MSBUILDDISABLENODEREUSE to 1 with no custom scripts and the node properly shuts down. Obviously the bigger issue here is a node being passed nodeReuse true. Running this a second time to be safe and I'll be fine merging it as-is and investigating further.

**Comment by Forgind** [comment]
When did we switch from waiting a week is ok to merging despite known issues?

**Comment by BenVillalobos** [comment]
> When did we switch from waiting a week is ok to merging despite known issues?

@Forgind What? We didn't. I want this in a state that is "acceptable to merge" should we decide that we want to so we can act on the perf data. My last comment can be interpreted as "The nodereuse thing looks like a bug, but setting the environment variable will allow us to act on this perf data while I dig into what the real issue here is".

**Comment by Forgind** [comment]
> I'll be fine merging it as-is and investigating further.

This sounded to me like merging early. If you meant to say that you think it's ready to merge but still intend to wait a week, then sure, though I'm less confident.

**Comment by BenVillalobos** [comment]
After mulling this over some, I don't think this should be blocked on the hanging process. It sheds light on a potential issue but isn't the cause.

Locally run `eng\cibuild_bootstrapped_msbuild.cmd` and see that there will always be a hanging process that was launched with `nodereuse:true` despite our build scripts passing `/nr:false`. I'll get an issue written up for it during PR review, but my best guess is that running our build scripts with `-test` causes nodereuse to be true for at least one node to help tests run faster.

**Comment by BenVillalobos** [comment]
Feedback from PR review: Maybe the hanging node is actually _spawned_ from a unit test. I didn't consider this and it sounds very plausible!

**Comment by BenVillalobos** [comment]
`StopOnFirstFailureandBuildInParallelMultipleNode`, `PropertyOverrideContainsSemicolon`, and all tests under `TargetsFile_Test` are the culprits. 

`StopOnFirstFailureandBuildInParallelMultipleNode` spawns a process with /nr:true (default for buildmanager)

`PropertyOverrideContainsSemicolon` and tests under `TargetsFile_Test` spawn VBCSCompiler.

**Comment by BenVillalobos** [comment]
Resolved the issue by disabling DOTNET_PERFLOG_DIR for all unit test assemblies. Created an issue tracking hanging assemblies here: https://github.com/dotnet/msbuild/issues/6344.

Confirmed locally that no extra perf logs get generated from tests. CI should succeed with no issues.

File: eng/cibuild_bootstrapped_msbuild.ps1
Comment (rainersigwald) [inline eng/cibuild_bootstrapped_msbuild.ps1:128]: Setting this at this layer means it applies only to the bootstrapped build. That's mostly fine, but it might be nice to have the perf data for the Windows/Release build too, which doesn't run the bootstrap. Worth changing? To argue with myself for a second, maybe not because it'd always be collecting the data from the LKG MSBuild installed on the build agent which is maybe not interesting.

**Review by rainersigwald — APPROVED** [review]

File: eng/cibuild_bootstrapped_msbuild.ps1
Comment (BenVillalobos) [reply eng/cibuild_bootstrapped_msbuild.ps1:128]: (I happened to check back in on this right as this was posted)

I considered that but ultimately came to the same conclusion. I can envision a future where we capture both and run a tool to diff the two so we can have our own mini-RPS, but I'll save that for when we decide to prioritize something like that.

**Pull request merged**
